include/hipSYCL/runtime/hardware.hpp

```diff
@@ -52,6 +52,9 @@ enum class device_uint_property {
   max_global_size1,
   max_global_size2,
   max_group_size,
+  max_num_sub_groups,
+  sub_group_independent_forward_progress,
+  sub_group_size,
```

Comment:
It might be better to not force this into the uint property API as we might have to return multiple values for this one. This is especially true for the new Level Zero/SPIR-V backend.
Add a dedicated virtual function for that?